tests/test_diffusers.py

```diff
@@ -1759,6 +1766,238 @@ def test_stable_diffusion_multicontrolnet_hpu_graphs(self):
+            image=self.get_dummy_image(),
+            generator=torch.Generator("cpu").manual_seed(0),
+            num_inference_steps=2,
+            output_type="np",
+        )
+        images = outputs.images
+
+        assert len(images) == 2
+        assert images[-1].shape == (32, 32, 3)
+
+    def test_depth2img_pipeline_bf16(self):
+        components = self.get_dummy_components()
+        gaudi_config = GaudiConfig(use_torch_autocast=True)
+
+        pipe = GaudiStableDiffusionDepth2ImgPipeline(
+            use_habana=True,
+            gaudi_config=gaudi_config,
+            **components,
+        )
+        pipe.set_progress_bar_config(disable=None)
+
+        outputs = pipe(
+            prompt="A painting of a squirrel eating a burger",
+            image=self.get_dummy_image(),
+            generator=torch.Generator("cpu").manual_seed(0),
+            num_inference_steps=2,
+            output_type="np",
+        )
+        images = outputs.images
+
+        assert len(images) == 1
+        assert images[0].shape == (32, 32, 3)
+
+    def test_depth2img_pipeline_hpu_graphs(self):
+        components = self.get_dummy_components()
+        gaudi_config = GaudiConfig(use_torch_autocast=False)
+
+        pipe = GaudiStableDiffusionDepth2ImgPipeline(
+            use_habana=True,
+            use_hpu_graphs=True,
+            gaudi_config=gaudi_config,
+            **components,
+        )
+        pipe.set_progress_bar_config(disable=None)
+
+        outputs = pipe(
+            prompt="A painting of a squirrel eating a burger",
+            image=self.get_dummy_image(),
+            generator=torch.Generator("cpu").manual_seed(0),
+            num_inference_steps=2,
+            output_type="np",
+        )
+        images = outputs.images
+
+        assert len(images) == 1
+        assert images[0].shape == (32, 32, 3)
+
+    @slow
+    def test_depth2img_pipeline_latency_bf16(self):
+        gaudi_config = GaudiConfig(use_torch_autocast=True)
+        model_name = "stabilityai/stable-diffusion-2-depth"
+        scheduler = GaudiDDIMScheduler.from_pretrained(model_name, subfolder="scheduler")
+
+        pipe = GaudiStableDiffusionDepth2ImgPipeline.from_pretrained(
+            model_name, gaudi_config=gaudi_config, scheduler=scheduler, use_habana=True, use_hpu_graphs=True
+        )
+
+        start_time = time.time()
+        outputs = pipe(
+            prompt="A painting of a squirrel eating a burger",
+            image=self.get_dummy_image((1, 3, 256, 256)),
+            generator=torch.Generator("cpu").manual_seed(0),
+            num_inference_steps=50,
+            output_type="np",
+        )
+        end_time = time.time()
+        latency = end_time - start_time
+        images = outputs.images
+
+        assert len(images) == 1
+        assert images[0].shape == (256, 256, 3)
+
+        assert latency < 1.05 * DEPTH2IMG_GENERATION_LATENCY_BASELINE_BF16
```

Comment:
Is there a way to test the fidelity of the image to the prompt?

something like this?
https://github.com/huggingface/optimum-habana/pull/920/files#diff-5d657a955747ad953e9a153bda1b8fd7324322d2e19c429c15e749cd9ed39f31

Reply:
The pipeline depends on an input image. Currently, it just draws random numbers for pixel values as a mock input. Are we allowed to add downloads for images? I don't think it is possible to test fidelity otherwise.

Reply:
I see. Not sure I can remember any other test that has an inp image, so not sure how that might work.

Reply:
Yes, you can download an image as part of the test. Here is an example: https://github.com/huggingface/optimum-habana/blob/6b2a5f843c963a9419ad246c77c3a39d671da198/tests/test_image_classification.py#L46

Reply:
I added fidelity testing on the prompt. Is this sufficient?

Reply:
It looks good to me. However, the latency assert fails on my Gaudi2 instance:
```
assert 38.6319305896759 < (1.05 * 28.13371205329895)
```
Does it consistently succeed on yours?

Reply:
It consistently passes for me after 12 attempts with no failures.

Reply:
Weird, I'll probably modify it later so that it passes on my instance.